Split package into client and daemon

hmaarrfk opened the pull request:
Closes #31 
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

hmaarrfk committed:
Split package

hmaarrfk committed:
MNT: Re-rendered with conda-build 3.23.3, conda-smithy 3.22.1, and conda-forge-pinning 2023.02.18.11.03.35

ryanvolz commented on recipe/meta.yaml:
```diff
@@ -13,76 +12,129 @@ source:
+  - name: pulseaudio
+    build:
+      run_exports:
+        - {{ pin_subpackage('pulseaudio', max_pin='x.x') }}
```
I think we are safe to `run_export` `pulseaudio-client` here from `pulseaudio` so that going forward packages building against libpulse will only pull in `pulseaudio-client` at runtime. Probably easier to do that than transition packages to directly build against `pulseaudio-client` and change the globally pinned package.

ryanvolz commented on recipe/meta.yaml:
```diff
@@ -13,76 +12,129 @@ source:
+        - pkg-config
+      host:
+        - alsa-lib
+        - alsa-lib >=1.0.24
```
It looks like the client doesn't need alsa to build, and the DSO analysis reports it as an overdependence.
```suggestion

```

ryanvolz commented on recipe/meta.yaml:
```diff
@@ -13,76 +12,129 @@ source:
+        - dbus >=1.4.12
+        - glib
+        - glib >=2.28.0
+        - jack
```
Same here, overdependence.
```suggestion

```

ryanvolz commented on recipe/meta.yaml:
```diff
@@ -13,76 +12,129 @@ source:
 about:
   home: https://www.freedesktop.org/wiki/Software/PulseAudio/
   license: LGPL-2.1-or-later
-  license_family: GPL
+  license_family: LGPL
   license_file: LGPL
```
Can/should we set the license to `GPL` for the `pulseaudio-daemon` output, to reflect building against `fftw`? I guess that would apply to the `pulseaudio` metapackage too?

hmaarrfk replied:
yes, we should..... i just need to find the syntax.

ryanvolz commented on recipe/meta.yaml:
```diff
@@ -13,76 +12,129 @@ source:
+      commands:
+        - test -f `pkg-config --variable=modlibexecdir --dont-define-prefix libpulse`/libprotocol-native.so
+        - pa-info
+        - pacat --help
```
Looks like `pacat` ends up in `pulseaudio-client`, move this test there instead.

ryanvolz commented on recipe/meta.yaml:
```diff
@@ -13,76 +12,129 @@ source:
+        - dbus >=1.4.12
+        - fftw
+        - fftw >=3.0
+        - glib
+        - glib >=2.28.0
```
DSO report shows that `libglib` is not used in `pulseaudio-daemon`, so maybe this can be removed? I still see configure output for it though, but that could be a red herring and it might configure fine without it.

ryanvolz commented on recipe/meta.yaml:
```diff
@@ -13,76 +12,129 @@ source:
+        - pa-info
+        - pacat --help
+        - pacmd --help
+        - pactl --help
```
Same with `pactl`, it looks like it is in `pulseaudio-client`.

ryanvolz reviewed (changes requested):
Looks pretty good! Just some potential tweaks left that I noticed.

hmaarrfk committed:
from jack and alsa from client

hmaarrfk committed:
Adjust tests

ryanvolz committed:
run_export pulseaudio-client from pulseaudio metapackage.

ryanvolz committed:
Try removing glib dep from pulseaudio-daemon.

ryanvolz committed:
Set license for pulseaudio-daemon output as GPLv2+.

It links with fftw which is GPLv2+, so that makes this output similarly
GPL-licensed.

ryanvolz committed:
MNT: Re-rendered with conda-build 3.23.3, conda-smithy 3.23.0, and conda-forge-pinning 2023.03.20.19.31.17

ryanvolz committed:
Revert "Try removing glib dep from pulseaudio-daemon."

This reverts commit a5a7aff95c6dfedc2dfe1322c52c63fc4f746800.
